base-permissions/src/iosMain/kotlin/PermissionRefreshScheduler.kt

```diff
@@ -21,21 +21,26 @@ import co.touchlab.stately.concurrency.AtomicBoolean
 import co.touchlab.stately.concurrency.AtomicReference
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.FlowCollector
 import kotlinx.coroutines.isActive
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 import kotlin.time.Duration
 
+interface AuthorizationStatusProvider {
```

Comment:
Does this always provide the Current status? In that case I would recommend `CurrentAuthorizationStatusProvider`